docker/docker_client.go

```diff
@@ -343,6 +343,7 @@ func SearchRegistry(ctx context.Context, sys *types.SystemContext, registry, ima
 	v1Res := &V1Results{}
 
 	// Get credentials from authfile for the underlying hostname
+	// nolint: staticcheck
```

Comment:
Yes, by adding a custom rules go `.golangci.yml`. Changed to use that instead of the comment.

Reply:
Ouch, that‘s worse than silencing the checker, and the ideal fix doesn’t work: https://github.com/golangci/golangci-lint/issues/741 

I have ended up with
```go
	//lint:ignore SA1019 We can't use GetCredentialsForRef because we want to search the whole registry.
	auth, err := config.GetCredentials(sys, registry) //nolint:staticcheck // https://github.com/golangci/golangci-lint/issues/741
```

so that both a direct invocation of `staticcheck` and use via `golangci-lint` don’t complain.

Reply:
Fixed as suggested, thank you for your suggestion.